graphql request in react

gabrielkay opened the pull request:
Sorry for the long PR, mainly graphql generated stuff.
- installed graphql, graphiql, apollo, gql
- expanded month_progress in Goal method to two methods, one that returns logs and one that returns a sum for use in views
- graphql request in app/javascript/packs/charts.js
- data formatting helper method in app/javascript/utils/processLogs.js
- slight tweaks to chart formatting
@solomonhawk

prayash commented on app/javascript/packs/charts.js:
```diff
@@ -1,13 +1,55 @@
+// var monthsAgo = 0;
+const logsQuery = gql`
+  query LogsQuery{
+    logs(goalId: 2, monthsAgo: 0) {
```
You could probably refactor this to take in variables if you see this becoming a dynamic query later on.

```js
const GET_LOGS = gql`
  query getLogs($id: ID!, $monthsAgo: Int!) {
      logs(goalId: $id, monthsAgo: $monthsAgo) {
         completedOn
         quantity
      }
  }
`
```

gabrielkay replied:
👍 that's the next thing I'm going to change, but I'll need to pass in id and monthsAgo as props which will take a bit of work so will do it next PR.

prayash commented on app/javascript/packs/charts.js:
```diff
@@ -1,13 +1,55 @@
-  const month = el.getAttribute("data-month")
-  ReactDOM.render(<MonthlyProgressChart data={data} label={label} month={month} />, el)
+client.query({
+  query: logsQuery
```
This is an interesting way of making a query, but I'd consider using Apollo's built-in `Query` component instead of a Promise based API. It's more idiomatic and will let you track loading and error states much easier. Up to you!
https://www.apollographql.com/docs/react/essentials/queries/

gabrielkay replied:
Did it this way to break up steps and find my errors but I see the way you linked is more common, I'll look into refactoring it that way this afternoon.

prayash commented on app/javascript/packs/charts.js:
```diff
@@ -1,13 +1,55 @@
 import ReactDOM from 'react-dom'
+import ApolloClient from "apollo-boost";
+import { InMemoryCache } from 'apollo-cache-inmemory';
+import { ApolloProvider } from 'react-apollo'
```
Looks like this isn't getting used. If you use the `<Query>` component you'll need to use `<ApolloProvider>` as the root component of the app though.

gabrielkay commented:
@prayash I kept the current query structure and added passing variables into the queries like you mentioned above. Also removed unused <ApolloProvider> import. I toyed with refactoring the query into the graph component but feel fine with how it's structured right now. Could I get a ✅ if it's looking good

solomonhawk commented on app/graphql/types/query_type.rb:
```diff
@@ -0,0 +1,23 @@
+      Goal.month_progress(goal_id, months_ago)
+    end
+
+# query {
```
🔪 ? It's fine to leave this as reference, but personally I'd remove it to keep this file nice and clean.

solomonhawk commented on app/javascript/utils/processLogs.js:
```diff
@@ -0,0 +1,53 @@
+  const daysInThisMonth = daysInMonth(month, year);
+
+  var sample = buildDate (year, month, 5);
+  console.log(sample);
```
🔪
